R/calculate_silhouette_width.R

```diff
@@ -0,0 +1,76 @@
+#' Calculate silhouette width scores from an integrated SCE object
+#'
+#' @param integrated_sce The integrated SCE object
+#' @param pc_name The name that allows access to the PCs. Example: fastMNN_PCA
+#' @param frac_cells The fraction of cells to downsample to. Default: 0.8
+#' @param nreps The number of times to repeat sub-sampling procedure. Default: 20
+#' @param seed Seed for initializing random sampling
+#' @param batch_column The variable in `integrated_sce` indicating the grouping of interest.
+#'  Generally this is either batches or cell types. Default is "library_id".
+#'
+#' @return Tibble with five columns: `rep`, representing the given downsampling replicate;
+#'   `silhouette_width`, the calculated silhouette width for the given `rep`; `silhouette_cluster`,
+#'   the assigned cluster for the cell during silhouette calculation, i.e. the true identity;
+#'   `other_cluster`, the other assigned for the cell during silhouette calculation;
+#'   `pc_name`, the name associated with the pc results
+#'
+#' @import SingleCellExperiment
+#'
+#' @export
+calculate_silhouette_width <- function(integrated_sce,
+                                       pc_name,
+                                       frac_cells = 0.8,
+                                       nreps = 20,
+                                       seed = NULL,
+                                       batch_column = "library_id") {
```

Comment:
Just a very minor suggestion to change default order here
```suggestion
                                       batch_column = "library_id",
                                       frac_cells = 0.8,
                                       nreps = 20,
                                       seed = NULL) {
```